### PR TITLE
fix: add REST mapping to proto files

### PR DESCRIPTION
**Pull request by sishbi**
fix: add REST mapping to proto files 
- use `buf` instead of `protoc` 
- create OpenAPI specs for new REST apis 
- generate an `image.binpb` for use by envoy

### CODE REVIEW

File: src/main/proto/com/kodypay/grpc/ecom/v1/ecom.proto [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
 
 // All service requests require X-API-Key header with 'API Key' value
 service KodyEcomPaymentsService {
   // Initiates a payment and returns a URL for the user to complete payment
-  rpc InitiatePayment(PaymentInitiationRequest) returns (PaymentInitiationResponse);
-  rpc PaymentDetails(PaymentDetailsRequest) returns (PaymentDetailsResponse);
-  rpc GetPayments(GetPaymentsRequest) returns (GetPaymentsResponse);
-  rpc GetCardToken(GetCardTokenRequest) returns (GetCardTokenResponse);
+  rpc InitiatePayment(PaymentInitiationRequest) returns (PaymentInitiationResponse) {
+    option (google.api.http) = {
```
Comment (sarah-jay-peper) [inline src/main/proto/com/kodypay/grpc/ecom/v1/ecom.proto:17]: the http config has to be backed into the proto?

**Review by sarah-jay-peper — APPROVED** [review]

File: src/main/proto/com/kodypay/grpc/ecom/v1/ecom.proto (hunk 1 above)
Comment (sishbi) [reply src/main/proto/com/kodypay/grpc/ecom/v1/ecom.proto:17]: Yes, but I think there is another way - I am still experimenting.

File: buf.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: v2
+modules:
+  - path: src/main/proto
+deps:
+  - buf.build/googleapis/googleapis
+lint:
```
Comment (rfontes17) [inline buf.yaml:6]: If we are going to use the lint, we could add an step on the pipe to validate we are following the basic conventions

Comment (sishbi) [reply buf.yaml:6]: It is automatically applied as part of the build, so will fail the build if the lint fails

Comment (rfontes17) [reply buf.yaml:6]: nice

**Review by ioannis-kody — APPROVED** [review]

**Pull request merged**